### DISCUSSION TITLE
[CI] Switch to community code formatting check.

### DISCUSSION THREAD

**Discussion by bader** [intel/llvm#12085] [Ideas]
LLVM project added code format checker - https://github.com/intel/llvm/blob/sycl/.github/workflows/pr-code-format.yml.
I suggest we remove `lint` check and GitHub Action we developed ourselves. We better contribute/extend community version than maintain our own version.
- pre-commit lint check - https://github.com/intel/llvm/blob/4ca27f8359e5bac7145c463849b3838b609163c8/.github/workflows/sycl_linux_precommit.yml#L33-L60
- GitHub Action - https://github.com/intel/llvm/tree/sycl/devops/actions/clang-format

@intel/dpcpp-devops-reviewers.

**Comment by aelovikov-intel** [comment]
It's running on `ubuntu-latest` (unless we customize just that piece of it), meaning we might have to wait a lot for its results. I also don't see how we can keep preventing regular CI from starting when the formatting task fails.

**Reply by AlexeySachkov** [reply]
> It's running on `ubuntu-latest` (unless we customize just that piece of it)

That seems like a reasonable enough downstream customization

> I also don't see how we can keep preventing regular CI from starting when the formatting task fails.

AFAIK, there was recently a change made to cancel in-progress CI in case when new commits are pushed to a PR - perhaps that is enough so we can proceed without blocking pre-commit on bad formatting?
